sway/config.c

```diff
@@ -53,53 +54,68 @@ static void free_mode(struct sway_mode *mode) {
 	free(mode);
 }
 
-void free_config(struct sway_config *config) {
+void free_config(struct sway_config *old_config) {
```

Comment:
This is pretty ugly but I need to access both old_config and the global variable 'config' inside this now; I'd usually be adding a second "optional" argument to compare to but that doesn't seem to be the style currently.
On final free_config at sway shutdown we won't be destroying seats (as they're all in the current config) but this is fine because the display going down should free stuff appropriately.
. . . now I'm writing that though this actually isn't fine for another reason, we aren't removing from the list as we process it so that should have been accessing freed names, I'll repush something to kludge that out, although this really makes me want to go for extra argument...